packages/storefront-lwc/src/modules/commerce/headerBasket/headerBasket.js

```diff
@@ -33,14 +32,17 @@ export default class HeaderBasket extends LightningElement {
         }
     }
 
-    constructor() {
-        super();
-        ShoppingBasket.updateBasketListener(
-            this.updateBasketHandler.bind(this),
+    renderedCallback() {
+        window.addEventListener(
+            'headerbasketcount',
+            event => (this.quantity = event.detail),
         );
-    }
 
-    updateBasketHandler() {
-        this.quantity = ShoppingBasket.basket.totalProductsQuantity || 0;
+        const qty = this.quantity;
+
+        window.addEventListener(
+            'removelineitem',
+            event => (this.quantity = qty - event.detail.quantity),
+        );
```

Comment:
Are we still using `window` for event handling?

Reply:
good catch, will remove it